src/components/ListNav/ListNav.tsx

```diff
@@ -27,32 +27,39 @@ import { AiOutlineArrowLeft } from "react-icons/ai";
 import { BackendInterface } from "services/BackendInterface";
 import { Ingredient } from "../../models/Ingredient";
+import { List } from "../../models/List";
 
 interface ListNavProps {
 	backendInterface: BackendInterface;
 	userAuth: UserAuth;
 }
 
 function ListNav({ userAuth, backendInterface }: ListNavProps) {
-	const { listName } = useParams<{ listName: string }>();
-	const navigate = useNavigate();
-	const [ingredients, setIngredients] = useState<Ingredient[]>([]);
-	const [open, setOpen] = useState(false); //first dialogue for all ingredients
-	const [allIngredients, setAllIngredients] = useState<Ingredient[]>([]);
-	const [chosenIngredient, setChosenIngredient] = useState<Ingredient | null>(null);
-	const [amount, setAmount] = useState<number | "">("");
-	const [units, setUnits] = useState<string[]>([]);
-	const [selectedUnit, setSelectedUnit] = useState<string>("g");
-	const [ingredientToDelete, setIngredientToDelete] = useState<Ingredient | null>(null);
-	const [openUnitDialog, setOpenUnitDialog] = useState(false);
-	const [openConfirmDialog, setOpenConfirmDialog] = useState(false);
-	const [ingredientToEdit, setIngredientToEdit] = useState<Ingredient | null>(null);
-	const [openEditDialog, setOpenEditDialog] = useState(false);
-	const [editAmount, setEditAmount] = useState<number | "">("");
-	const [editUnit, setEditUnit] = useState<string>("g");
-	const [amountError, setAmountError] = useState<string>("");
-	const [searchQuery, setSearchQuery] = useState<string>("");
-	const [dialogSearchQuery, setDialogSearchQuery] = useState<string>("");
+    const { listName } = useParams<{ listName: string }>();
+    const navigate = useNavigate();
+    const [ingredients, setIngredients] = useState<Ingredient[]>([]);
+    const [open, setOpen] = useState(false); //first dialogue for all ingredients
+    const [allIngredients, setAllIngredients] = useState<Ingredient[]>([]);
+    const [chosenIngredient, setChosenIngredient] = useState<Ingredient | null>(null);
+    const [amount, setAmount] = useState<number | ''>('');
+    const [units, setUnits] = useState<string[]>([]);
+    const [selectedUnit, setSelectedUnit] = useState<string>("g");
+    const [ingredientToDelete, setIngredientToDelete] = useState<Ingredient | null>(null); 
+    const [openUnitDialog, setOpenUnitDialog] = useState(false);
+    const [openConfirmDialog, setOpenConfirmDialog] = useState(false); 
+    const [ingredientToEdit, setIngredientToEdit] = useState<Ingredient | null>(null);
+    const [openEditDialog, setOpenEditDialog] = useState(false);
+    const [editAmount, setEditAmount] = useState<number | ''>('');
+    const [editUnit, setEditUnit] = useState<string>("g");
+    const [amountError, setAmountError] = useState<string>('');
+    const [searchQuery, setSearchQuery] = useState<string>("");
+    const [dialogSearchQuery, setDialogSearchQuery] = useState<string>("");
+    const [userLists, setUserLists] = useState<List[]>([]); // To store the user's lists
```

Comment:
Sorry, another real quick thing. This state variable is not used anywhere. could we remove it please? Other than that, everything else looks great!